kubernetes/plausible.yaml

```diff
@@ -0,0 +1,101 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: plausible
+  labels:
+    app.kubernetes.io/name: plausible
+    app.kubernetes.io/component: server
+spec:
+  type: LoadBalancer
+  ports:
+    - name: http
+      port: 8000
+      targetPort: 8000
+      protocol: TCP
+  selector:
+    app.kubernetes.io/name: plausible
+    app.kubernetes.io/component: server
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: plausible
+  labels:
+    app.kubernetes.io/name: plausible
+    app.kubernetes.io/component: server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: plausible
+      app.kubernetes.io/component: server
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: plausible
+        app.kubernetes.io/component: server
+    spec:
+      restartPolicy: Always
+      containers:
+        - name: plausible
+          image: plausible/analytics:latest
+          imagePullPolicy: Always
+          args:
+            - /bin/sh
+            - -c
+            - sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh db init-admin && /entrypoint.sh run
+          ports:
+            - containerPort: 8000
+          envFrom:
+            - configMapRef:
+                name: plausible-config
+          env:
+            - name: POSTGRES_USER
+              valueFrom:
+                secretKeyRef:
+                  name: plausible-db-user
+                  key: username
+            - name: POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: plausible-db-user
+                  key: password
+            - name: CLICKHOUSE_USER
+              valueFrom:
+                secretKeyRef:
+                  name: plausible-events-db-user
+                  key: username
+            - name: CLICKHOUSE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: plausible-events-db-user
+                  key: password
+            - name: DATABASE_URL
+              value: postgres://$(POSTGRES_USER):$(POSTGRES_PASSWORD)@$(PLAUSIBLE_DB_SERVICE_HOST):$(PLAUSIBLE_DB_SERVICE_PORT)/plausible
+            - name: CLICKHOUSE_DATABASE_URL
+              value: http://$(CLICKHOUSE_USER):$(CLICKHOUSE_PASSWORD)@$(PLAUSIBLE_EVENTS_DB_SERVICE_HOST):$(PLAUSIBLE_EVENTS_DB_SERVICE_PORT)/plausible
+            - name: SMTP_HOST_ADDR
+              value: $(PLAUSIBLE_SMTP_SERVICE_HOST)
+          securityContext:
+            allowPrivilegeEscalation: false
+          resources:
+            limits:
+              memory: 2Gi
+              cpu: 1500m
+            requests:
+              memory: 200Mi
+              cpu: 250m
```

Comment:
I think those are pretty beefy requirements for a minimal deployment. I guess it's good practice to include these but maybe lower the request a bit?

As a reference, the resource consumption of my deployments without any resource quotas:

```text
$ kubectl get pods -n plausible
NAME                                READY   STATUS    RESTARTS   AGE
event-data-sts-0                    1/1     Running   0          60d
plausible-deploy-5674c66c98-5q48w   1/1     Running   0          60d
$ kubectl top pod -n plausible
NAME                                CPU(cores)   MEMORY(bytes)
event-data-sts-0                    7m           65Mi
plausible-deploy-5674c66c98-5q48w   19m          146Mi
```

Reply:
It's probably good to not define or to set really low Pod resources, so anyone could bring it up on say kind or Minikube

Reply:
I didn't have access to `kubectl top` when working out the resource requirements and it would seem the commands I used to work it out were way off. I have lowered the requirements to what the stack is using at idle as reported by the Kubernetes Dashboard.